### ISSUE TITLE
Instruction on DCLM-Baseline reproduction and Filtering track

### ISSUE THREAD

**Issue by chenweize1998** [mlfoundations/dclm#44]
Hi. First of all, thank you for your great work. The DCLM framework appears to be a valuable tool for researchers to validate ideas related to training dataset curation. I have two main questions regarding the framework and its usage:

1. Could you please provide guidance on how to conduct experiments for the filtering track? I didn't find it in the README, perhaps I missed something? Specifically should we first download the DCLM dataset from [huggingface](https://huggingface.co/datasets/mlfoundations/dclm-baseline-1.0)? After downloading, do we then perform our filtering on this dataset?

2. I noticed that the example provided in the [README](https://github.com/mlfoundations/dclm/blob/main/baselines/README.md#running-the-processing-pipeline) seems to be based on the C4 dataset. I'm interested in reproducing the DCLM-Baseline results. In this regard, is there a specific guide or set of instructions for reproducing the DCLM-Baseline results? There are plenty of files in `exp_data/datasets/raw_sources`, which is kind of confusing.

Any clarification on these points would be greatly appreciated. Thank you!

**Comment by afang-story** [comment]
1. You should determine which scale you want to participate at, and download the corresponding pool:
- https://huggingface.co/datasets/mlfoundations/dclm-pool-400m-1x
- https://huggingface.co/datasets/mlfoundations/dclm-pool-1b-1x
- https://huggingface.co/datasets/mlfoundations/dclm-pool-1b-5x
- https://huggingface.co/datasets/mlfoundations/dclm-pool-7b-1x
- https://huggingface.co/datasets/mlfoundations/dclm-pool-7b-2x
2. For reproducing DCLM-baseline, you would run baselines/baselines_configs/dclm_baseline_refinedweb.yaml, some version of deduplication, and then baselines/baselines_configs/fasttext_filter.yaml

**Comment by afang-story** [comment]
Could you be more specific on what your questions are? The readme goes through how to run the individual steps of the pipeline, and the paper contains details on what choice to make at each step. Additionally, these depend on what exactly you want to reproduce.

For a brief overview see below. Note that DCLM baseline produces trillions of tokens, which has slight modifications from what was done for the leaderboard. To match what is on the leaderboard, you should run: 
- refined web processing (https://github.com/mlfoundations/dclm/blob/main/baselines/baselines_configs/refinedweb.yaml),
- minhash + exact dedup (https://github.com/mlfoundations/dclm/tree/main/dedup)
- fasttext processing (https://github.com/mlfoundations/dclm/blob/main/baselines/baselines_configs/fasttext_filter.yaml). 

You can then tokenize shuffle to produce the actual tokens that you train on. Please see our paper for additional details.

As detailed in "Selecting Raw Sources" you can see https://github.com/mlfoundations/dclm/tree/main/exp_data/datasets for examples of what jsons look like, and you may want to consider creating one for the new source. Whenever you run one of the steps we provide, a json for that output should also be created, which can then be used for the next step.

**Comment by chenweize1998** [comment]
Thank you for the additional clarification. It is now much clearer for me. However, upon following the steps you outlined, I noticed a few items missing from the current repository:

1. Banlist:
   The banlist referenced in this file is not present:
   https://github.com/mlfoundations/dclm/blob/c59dd5878c8bd1965b4894e20d8a091e8fdeac7a/baselines/baselines_configs/refinedweb.yaml#L5-L8

2. Documentation and Code:
   The README and source code for minhash and exact deduplication are absent.

Would it be possible to include these missing components? It would be of great help. Thanks.

**Comment by jeffreywpli** [comment]
The banlist download happens via our setup.py, as discussed in https://github.com/mlfoundations/dclm/issues/55 .

For dedup, we currently have our bff included and documented (which we used in the final version of DCLM-Baseline). The minhash and exact dedups will be added in the next couple days.
